src/Aspire.Hosting.Azure.AppContainers/AzureContainerAppsInfrastructure.cs

```diff
@@ -70,23 +72,50 @@ public async Task BeforeStartAsync(DistributedApplicationModel appModel, Cancell
         {
             appModel.Resources.Add(additionalModule);
         }
+
+        static void SetKnownParameterValue(AzureBicepResource r, string key, Func<AzureBicepResource, object> factory)
+        {
+            if (r.Parameters.TryGetValue(key, out var existingValue) && existingValue is null)
+            {
+                var value = factory(r);
+
+                if (value is BicepOutputReference b && b.Resource == r)
+                {
+                    return;
+                }
+
+                r.Parameters[key] = value;
+            }
+        }
+
+        // Resolve the known parameters for the container app environment
+        foreach (var r in appModel.Resources.OfType<AzureBicepResource>())
+        {
+            // HACK: This forces parameters to be resolved for any AzureProvisioningResource
+            r.GetBicepTemplateFile();
```

Comment:
🤮. What are the downsides of doing this here and again when really generating it?

Reply:
In theory, you can add more parameters in another before start hook.